### PR TITLE
LPD-18769 Workflow definition that use groovy script in any part of its content should not be published when script content are not allowed on its system

### PR DESCRIPTION
**Pull request by MarinhoFeliphe**

### CODE REVIEW

File: modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/runtime/integration/test/WorkflowDefinitionManagerTest.java [hunk 1]
```diff
@@ -52,6 +59,26 @@ public void testDeleteSaveWorkflowDefinition() throws Exception {
 			workflowDefinition.getVersion());
 	}
 
+	@Test
+	public void testDeployGroovyScriptBasedWorkflowDefinition()
```
Comment (guilhermedcamacho) [inline modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/runtime/integration/test/WorkflowDefinitionManagerTest.java:63]: Do you think it's worth putting this method in some util class and reusing it here and in the **ObjectActionLocalServiceTest** and **ObjectValidationRuleLocalServiceTest** classes? We are replicating this method in the test classes.

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

File: modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/runtime/integration/test/WorkflowDefinitionManagerTest.java [hunk 2]
```diff
@@ -503,6 +530,29 @@ private void _assertValid(InputStream inputStream) throws Exception {
 			FileUtil.getBytes(inputStream));
 	}
 
+	private Closeable _disableScriptContentBeExecutedOrIncluded()
```
Comment (guilhermedcamacho) [inline modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/runtime/integration/test/WorkflowDefinitionManagerTest.java:533]: Do you think it's worth putting this method in some util class and reusing it here and in the ObjectActionLocalServiceTest and ObjectValidationRuleLocalServiceTest classes? We are replicating this method in the test classes.

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

File: modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/runtime/integration/test/WorkflowDefinitionManagerTest.java (hunk 2 above)
Comment (MarinhoFeliphe) [reply modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/runtime/integration/test/WorkflowDefinitionManagerTest.java:533]: I'll send a follow-up pull request with this change,

Thanks Camacho!

Comment (guilhermedcamacho) [reply modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/runtime/integration/test/WorkflowDefinitionManagerTest.java:533]: No problem, I'll mark this as resolved and approve this PR then.

**Comment by MarinhoFeliphe** [comment]
Resent here: https://github.com/brianchandotcom/liferay-portal/pull/148705

**Pull request closed without merging**